routers/api/v2/users_test.go

```diff
@@ -0,0 +1,253 @@
+package v2
+
+import (
+	"errors"
+	"github.com/gin-gonic/gin"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/unicsmcr/hs_auth/config"
+	"github.com/unicsmcr/hs_auth/entities"
+	mock_v2 "github.com/unicsmcr/hs_auth/mocks/authorization/v2"
+	mock_services "github.com/unicsmcr/hs_auth/mocks/services"
+	mock_utils "github.com/unicsmcr/hs_auth/mocks/utils"
+	"github.com/unicsmcr/hs_auth/services"
+	"github.com/unicsmcr/hs_auth/testutils"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.uber.org/zap"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+const testAuthTokenLifetime = 10000000
+
+type usersTestSetup struct {
+	ctrl             *gomock.Controller
+	router           APIV2Router
+	mockUService     *mock_services.MockUserService
+	mockAuthorizer   *mock_v2.MockAuthorizer
+	mockTimeProvider *mock_utils.MockTimeProvider
+	testUser         *entities.User
+	testCtx          *gin.Context
+	w                *httptest.ResponseRecorder
+}
+
+func setupUsersTest(t *testing.T) *usersTestSetup {
+	ctrl := gomock.NewController(t)
+	mockAuthorizer := mock_v2.NewMockAuthorizer(ctrl)
+	mockUService := mock_services.NewMockUserService(ctrl)
+	mockTimeProvider := mock_utils.NewMockTimeProvider(ctrl)
+
+	router := NewAPIV2Router(zap.NewNop(), &config.AppConfig{
+		AuthTokenLifetime: testAuthTokenLifetime,
+	}, mockAuthorizer, mockUService, mockTimeProvider)
+
+	w := httptest.NewRecorder()
+	testCtx, _ := gin.CreateTestContext(w)
+
+	testUser := entities.User{
+		ID:       primitive.NewObjectID(),
+		Name:     "Bob the Tester",
+		Email:    "test@email.com",
+		Team:     primitive.NewObjectID(),
+		Password: "password123",
+	}
+
+	return &usersTestSetup{
+		ctrl:             ctrl,
+		router:           router,
+		mockUService:     mockUService,
+		mockAuthorizer:   mockAuthorizer,
+		mockTimeProvider: mockTimeProvider,
+		testUser:         &testUser,
+		testCtx:          testCtx,
+		w:                w,
```

Comment:
What does `w` stand for?

Reply:
writer in `ResponseWriter`:
https://godoc.org/github.com/gin-gonic/gin#CreateTestContext